torch/_dynamo/variables/builtin.py

```diff
@@ -515,46 +536,68 @@ def call_function(
                         *proxy_args_kwargs(args, kwargs),
                     )
 
-                    return wrap_fx_proxy_cls(variables.NumpyNdarrayVariable, tx, proxy)
+                    ret = wrap_fx_proxy_cls(variables.NumpyNdarrayVariable, tx, proxy)
+                    if fn in self._self_assigning_ops() and isinstance(
+                        args[0], variables.TensorVariable
+                    ):
+                        assert (
+                            args[0].as_proxy().node.meta["example_value"]
+                            is ret.as_proxy().node.meta["example_value"]
+                        )
```

Comment:
Do the inverse assert? (If it's not self assigning, then it shouldn't "is" match?)